pages/model/[uuid].tsx

```diff
@@ -111,17 +116,60 @@ function Model() {
   }
 
   useEffect(() => {
-    if (tab !== undefined && isTabOption(tab)) {
-      setGroup(tab)
+    if (currentUser && version) {
+      setIsManager(currentUser.id === version.metadata.contacts.manager)
+      setIsReviewer(currentUser.id === version.metadata.contacts.reviewer)
     }
-  }, [tab])
+  }, [currentUser, version])
+
+  const updateLastViewed = useCallback(
+    (role: string) => {
+      if (isManager && version?.updatedAt) {
```

Comment:
If you change this to `if (isManager && version && version.updatedAt)` then you can remove the null access operators `?` and non-null assertions `!` within the code block below

Reply:
This works for the content of the useEffect, but not the dependencies